src/charmhub_listing_review/update_issue.py

```diff
@@ -0,0 +1,458 @@
+        ['gh', 'issue', 'view', str(issue_number), '--json', 'body', '--jq', '.body'],  # noqa: S607
+        capture_output=True,
+        text=True,
+        check=True,
+    )
+    body = result.stdout
+
+    # Define the fields to extract and their headings
+    fields = {
+        'name': '### Charm name',
+        'demo_url': '### Demo',
+        'project_repo': '### Project Repository',
+        'ci_linting': '### CI Linting',
+        'ci_release_url': '### CI Release',
+        'ci_integration_url': '### CI Integration Tests',
+        'documentation_link': '### Documentation Link',
+    }
+
+    # Extract values for each field
+    issue_data: dict[str, bool | str | None] = {}
+    for key, heading in fields.items():
+        pattern = rf'{re.escape(heading)}\s*\n([^\n]*)'
+        match = re.search(pattern, body)
+        if match:
+            value = match.group(1).strip()
+            issue_data[key] = value
+        else:
+            issue_data[key] = None
+
+    # These have expected filenames, so we use those rather than require the author provide them.
+    # This is quite specific to GitHub, but we can add support for other platforms if required,
+    # and if they aren't found then the reviewer just has to locate them themselves.
+    issue_data['contribution_link'] = f'{issue_data["project_repo"]}/blob/main/CONTRIBUTING.md'
+    issue_data['license_link'] = f'{issue_data["project_repo"]}/blob/main/LICENSE'
+    issue_data['security_link'] = f'{issue_data["project_repo"]}/blob/main/SECURITY.md'
+
+    return cast('_IssueData', issue_data)
+
+
+def assign_review(issue_number: int, dry_run: bool=False):
+    """Assign the issue to a team.
+
+    We assign the issue to a single person (generally the manager) from a
+    charming team. The expectation is that they will then delegate the actual
+    review to a member of their team. The manager has overall responsibility for
+    ensuring that the review is completed on time (with Charm Tech responsible
+    for keeping the manager accountable).
+
+    The manager can't assign the issue directly to a reviewer (without having
+    every possible reviewer added as a collaborator on the repository), so they
+    are expected to simply ping them in a comment. Once they have submitted
+    their review, the author can interact with them in the usual way.
+    """
+    reviewers_file = pathlib.Path(__file__).parent.parent.parent / 'reviewers.yaml'
+    with reviewers_file.open('r') as f:
+        reviewers_data = yaml.safe_load(f)
+    reviewers = reviewers_data['reviewers']
+    teams = [info['team'] for info in reviewers.values()]
+    team = random.choice(teams)  # noqa: S311
+    # If there happen to be multiple people in a team, then randomly pick among
+    # them.
+    team_reviewers = [name for name, info in reviewers.items() if info['team'] == team]
+    reviewer = random.choice(team_reviewers)  # noqa: S311
+
+    if not dry_run:
+        subprocess.run(  # noqa: S603
```

Comment:
These subprocess-related noqa's just seem like noise. Should we just disable them? It seems doing it in pyproject.toml at the repo level would be reasonable for this repo.